frame/uniques/src/tests.rs

```diff
@@ -633,3 +644,45 @@ fn cancel_approval_works_with_force() {
 		);
 	});
 }
+
+#[test]
+fn max_supply_should_work() {
+	new_test_ext().execute_with(|| {
+		let collection_id = 0;
+		let user_id = 1;
+		let max_supply = 2;
+
+		// validate set_collection_max_supply
+		assert_ok!(Uniques::force_create(Origin::root(), collection_id, user_id, true));
+		assert!(!CollectionMaxSupply::<Test>::contains_key(collection_id));
+
+		assert_ok!(Uniques::set_collection_max_supply(
+			Origin::signed(user_id),
+			collection_id,
+			max_supply
+		));
+		assert_eq!(CollectionMaxSupply::<Test>::get(collection_id).unwrap(), max_supply);
+
+		assert!(events().contains(&Event::<Test>::CollectionMaxSupplySet {
```

Comment:
You can also use `assert_eq!(events().last().unwrap(), …` here to ensure that it is the last event.